### PR TITLE
APP-89 [C] Cross-compilation of the data collection demo

### PR DESCRIPTION
**Pull request by Dubland**
/cc @RostakaGmfun @vadimol @Slash32 

TBD: Check demo work with cc32xx and esp8266 after fixing encryption for this platforms.

### CODE REVIEW

File: datacollectiondemo/source/c/CMakeLists.txt [hunk 1]
```diff
@@ -15,22 +15,41 @@
...
+if (${KAA_PRODUCE_BINARY})
+    add_custom_command(TARGET ${APP_NAME} POST_BUILD
+            COMMAND ${CMAKE_OBJCOPY} -O binary ${APP_NAME} ${APP_NAME}.bin)
+endif ()
 
-set(APP_NAME "demo_client")
-set(KAA_SDK_DIR "client-c")
-
-execute_process(COMMAND ${CMAKE_COMMAND} -E make_directory ${KAA_SDK_DIR})
-execute_process(COMMAND sh -c "tar -zxf ${CMAKE_CURRENT_SOURCE_DIR}/libs/kaa/kaa-c*.tar.gz -C ${KAA_SDK_DIR}")
```
Comment (RostakaGmfun) [inline datacollectiondemo/source/c/CMakeLists.txt:25]: This probably should be left with some modifications.
Let's set `KAA_SDK_DIR` to `${CMAKE_CURRENT_LIST_DIR}/libs/kaa`. Additional check whether SDK has already been unpacked is also preferred.

File: datacollectiondemo/source/c/build.sh [hunk 2]
```diff
@@ -33,22 +34,87 @@ fi
...
+          -DKAA_PRODUCE_BINARY=$KAA_PRODUCE_BINARY \
+          -DWIFI_SSID=$SSID \
+          -DWIFI_PASSWORD=$PASSWORD \
+          -DCMAKE_BUILD_TYPE=MinSizeRel \
+          -DWITH_EXTENSION_EVENT=OFF \
           -DWITH_EXTENSION_CONFIGURATION=OFF \
           -DWITH_EXTENSION_NOTIFICATION=OFF \
           -DKAA_MAX_LOG_LEVEL=3 \
-          $KAA_TOOLCHAIN_PATH_SDK ..
+          ${KAA_TOOLCHAIN_PATH_SDK} ..
```
Comment (RostakaGmfun) [inline datacollectiondemo/source/c/build.sh]: Add quotes

**Comment by Slash32** [comment]
@kaa-leeroy test this please

File: datacollectiondemo/source/c/CMakeLists.txt (hunk 1 above)
Comment (RostakaGmfun) [reply datacollectiondemo/source/c/CMakeLists.txt:25]: Discussed

**Comment by Slash32** [comment]
@kaa-leeroy test this please

**Comment by Slash32** [comment]
LGTM

**Pull request merged**